user-service/src/main/java/hcmus/alumni/userservice/controller/UserServiceController.java

```diff
@@ -484,12 +513,602 @@ public ResponseEntity<HashMap<String, Object>> getSearchResult(
+		Optional<UserModel> optionalUser = userRepository.findById(id);
+
+		if (optionalUser.isEmpty()) {
+			throw new AppException(21501, "Không tìm thấy thông tin người dùng", HttpStatus.NOT_FOUND);
+		}
+		UserModel user = optionalUser.get();
+		if (avatarUrl != null && !avatarUrl.isEmpty()) {
+			try {
+				String avatarPath = imageUtils.saveImageToStorage(imageUtils.getAvatarPath(id), avatarUrl, "avatarUrl");
+				user.setAvatarUrl(avatarPath);
+			} catch (IOException e) {
+				e.printStackTrace();
+				throw new AppException(21502, "Lỗi khi tải ảnh lên", HttpStatus.BAD_REQUEST);
+			}
+		}
+		userRepository.save(user);
+		return ResponseEntity.status(HttpStatus.OK).body("");
+	}
+
+	@PutMapping("/profile/{id}/cover")
+	public ResponseEntity<String> updateProfileCover(@PathVariable String id,
+			@RequestParam(value = "coverUrl", required = true) MultipartFile coverUrl) {
+		Optional<UserModel> optionalUser = userRepository.findById(id);
+
+		if (optionalUser.isEmpty()) {
+			throw new AppException(21601, "Không tìm thấy thông tin người dùng", HttpStatus.NOT_FOUND);
+		}
+		UserModel user = optionalUser.get();
+		if (coverUrl != null && !coverUrl.isEmpty()) {
+			try {
+				String coverPath = imageUtils.saveImageToStorage(imageUtils.getCoverPath(id), coverUrl, "coverUrl");
+				user.setAvatarUrl(coverPath);
+			} catch (IOException e) {
+				e.printStackTrace();
+				throw new AppException(21602, "Lỗi khi tải ảnh lên", HttpStatus.BAD_REQUEST);
+			}
+		}
+		userRepository.save(user);
+		return ResponseEntity.status(HttpStatus.OK).body("");
+	}
+
+	@GetMapping("profile/{id}/job")
+	public ResponseEntity<Map<String, Object>> getAllJobs(@PathVariable String id) {
+		Optional<UserModel> optionalUser = userRepository.findById(id);
+		if (optionalUser.isEmpty()) {
+			throw new AppException(21701, "Không tìm thấy thông tin người dùng", HttpStatus.NOT_FOUND);
+		}
+
+		List<JobModel> jobs = jobRepository.findByUserId(id);
+		if (jobs.isEmpty()) {
+			throw new AppException(21702, "Không có công việc nào được tìm thấy cho người dùng này",
+					HttpStatus.NOT_FOUND);
+		}
+
+		HashMap<String, Object> result = new HashMap<>();
+		result.put("jobs", jobs);
+		return ResponseEntity.status(HttpStatus.OK).body(result);
+	}
+
+	@GetMapping("profile/{id}/job/{jobId}")
+	public ResponseEntity<IJobDto> getJobById(@PathVariable String id, @PathVariable String jobId) {
+		Optional<UserModel> optionalUser = userRepository.findById(id);
+		if (optionalUser.isEmpty()) {
+			throw new AppException(21801, "Không tìm thấy thông tin người dùng", HttpStatus.NOT_FOUND);
+		}
+
+		Optional<IJobDto> optionalJob = jobRepository.findByJobId(jobId);
+		if (optionalJob.isEmpty()) {
+			throw new AppException(21802, "Không có công việc nào được tìm thấy cho người dùng này",
+					HttpStatus.NOT_FOUND);
+		}
+		return ResponseEntity.status(HttpStatus.OK).body(optionalJob.get());
+	}
+
+	@PostMapping("profile/{id}/create-job")
```

Comment:
```suggestion
	@PostMapping("/profile/job")
```
bản thân method POST có nghĩa là tạo r
vì có userId trong request header r nên lược bỏ dc `{id}` trên url